src/etaoin/api.clj

```diff
@@ -1611,6 +1611,108 @@
   [driver q]
   (get-element-property driver q :value))
 
+(defn get-element-shadow-root-el
+  "Returns the shadow root for the specified element or `nil` if the
+  element does not have a shadow root."
+  [driver el]
+  (-> (get-element-property-el driver el "shadowRoot")
+      first
+      second))
+
+(defn get-element-shadow-root
+  "Returns the shadow root for the first element matching the query, or
+  `nil` if the element does not have a shadow root."
```

Comment:
I think in all other fns with `q` I've included:

See [[query]] for details on `q`.